src/data_structures/sparse_table.rs

```diff
@@ -0,0 +1,82 @@
+/*
+    A Sparse Table, is a data structure for answering range-minimum-queries of an array.
+    For a given array A[], of elements for which an ordering exists, we want to find the
+    minimum value A[x] of a subarray A[i..j], where i and j are the query parameters.
+
+    Precomputation complexity: O(n log(n))
+    Query complexity: O(1)
+
+    Wikipedia: <https://en.wikipedia.org/wiki/Range_minimum_query>
+*/
+
+use std::cmp::PartialOrd;
+
+pub struct SparseTable<T: PartialOrd + Copy> {
```

Comment:
My reason for naming it `SparseTable` is so that there is no confusion down the line, when other range-minimum-query data structures are added.
I worked on an implementation of an O(n) construction and O(1) query time solution and also wanted to contribute that.

Reply:
Done